# DeepSplice
A deep learning based splice junction sequence classifier.

Yi Zhang (yi dot zhang at uky dot edu)

***
## Table of Contents
* Introduction
* Prerequisites
* Data
***

## Introduction
Alternative splicing (AS) is a regulated cellular process that enables the production of multiple mRNA transcripts from a single multi-exon gene. The availability of large-scale RNA-seq datasets has made it possible to discover novel splice junctions through read alignments that extend across adjoining exons. This greatly enhances the capability to decipher gene structures and explore the diversity of splicing variants. However, existing ab initio aligners are vulnerable to false positive spliced alignments because of sequence errors and random sequence matches. These spurious alignments can lead to a significant set of false positive splice junction predictions, confusing downstream analyses of splice variant detection and abundance estimation. In this work, we present the first deep learning based splice junction sequence classifier, named DeepSplice, which employs convolutional neural networks to classify candidate splice junctions. We show (i) DeepSplice outperforms state-of-the-art methods for splice site classification when applied to the popular benchmark dataset HS3D, (ii) DeepSplice shows high accuracy for splice junction classification with GENCODE annotation, and (iii) the application of DeepSplice to classify putative splice junctions generated by Rail-RNA alignment of 21,504 human RNA-seq data significantly reduces 43 million candidates into around 3 million highly confident novel splice junctions.

## Prerequisites
* [Python](https://www.python.org/)
* [TensorFlow](https://www.tensorflow.org/)
* [NumPy](http://www.numpy.org/)

## Data
Splice junction classification result with [intropolis](http://intropolis.rail.bio) : a tab-delimited text file ([download posSJ.txt](https://drive.google.com/drive/folders/0B3-yEUAQDhwBU2JGSnFDV2hfSEk?usp=sharing)) with fields
  1. chromosome
  2. exon end position (1-based; inclusive)
  3. exon start position (1-based; inclusive)
  4. strand (+ or -)
  5. donor and acceptor dinucleotide (e.g., GTAG)
  6. number of samples in which junction was found
  7. number of reads mapping across junction in samples from field 6
  8. probability of being true

## Citation
Zhang, Y., Liu, X., MacLeod, J. and Liu, J., 2018. Discerning novel splice junctions derived from RNA-seq alignment: a deep learning approach. *BMC genomics, 19*(1), p.971.

DOI: [https://doi.org/10.1186/s12864-018-5350-1](https://doi.org/10.1186/s12864-018-5350-1)


